### ISSUE TITLE
Breaking API change between 1.0.2 and 1.0.3

### ISSUE THREAD

**Issue by pscanf** [unshiftio/url-parse#15]
I noticed a breaking change in the `query` property behaviour between `v1.0.2` and `v1.0.3`.
Reproduction:
```js
var Url = require("url-parse");
var url = new Url("http://github.com", true);
url.query.testKey = "testValue";
console.log(url.toString());
```
[JSBin showing behaviour for `1.0.2`](http://jsbin.com/setozevaxe/1/edit?html,js,console)
[JSBin showing behaviour for `1.0.3`](http://jsbin.com/xibiqexaku/edit?html,js,console)

When the url which gets parsed doesn't have a query string, the `query` property gets initialized to the empty string `""`, rather than to an empty object `{}`.

I suggest publishing a `1.0.4` version which reverts to the old behaviour, and, if the new behaviour is the intended one, publishing a `1.1.0` with that change.
I don't have time right now to prepare a PR, but I might do it tonight.

Cheers :wink:

**Comment by lpinca** [comment]
Thank you for reporting this issue.

**Comment by 3rd-Eden** [comment]
Released 1.0.4 with the bug fix.

**Comment by pscanf** [comment]
:+1: wonderful, thanks!
